auth/src/main/java/com/firebase/ui/auth/ui/idp/CredentialSignInHandler.java

```diff
@@ -69,24 +69,27 @@ public void onComplete(@NonNull Task<AuthResult> task) {
         } else {
             if (task.getException() instanceof FirebaseAuthUserCollisionException) {
                 final String email = mResponse.getEmail();
-                mHelper.getFirebaseAuth()
-                        .fetchProvidersForEmail(email)
-                        .addOnFailureListener(new TaskFailureLogger(
-                                TAG, "Error fetching providers for email"))
-                        .addOnSuccessListener(new StartWelcomeBackFlow(email))
-                        .addOnFailureListener(new OnFailureListener() {
-                            @Override
-                            public void onFailure(@NonNull Exception e) {
-                                // TODO: What to do when signing in with Credential fails
-                                // and we can't continue to Welcome back flow without
-                                // knowing providers?
-                            }
-                        });
+                if (email != null) {
```

Comment:
It's somewhat surprising that we ever got null email here but still managed to get a `CollisionException` ... but anyway this `null` check is still the right thing to do!